### PR TITLE
Add tests for MultiChainProvider and some small fixes

### PR DESCRIPTION
**Pull request by Mrtenz**
This adds tests for the `MultiChainProvider` to achieve 100% test coverage, and fixes some small issues in the provider.

### CODE REVIEW

File: packages/provider/src/MultiChainProvider.ts [hunk 1]
```diff
@@ -112,26 +131,33 @@ export class MultiChainProvider extends SafeEventEmitter implements Provider {
...
   }
 
+  /**
+   * Get the provider that is injected by the wallet, i.e., `window.ethereum`.
+   *
+   * @returns The injected provider.
+   */
   #getProvider() {
-    return window.ethereum;
+    return ethereum;
```
Comment (FrederikBolding) [inline packages/provider/src/MultiChainProvider.ts:140]: Is this safe 🤔

File: packages/provider/package.json [hunk 2]
```diff
@@ -28,7 +28,7 @@
     "@metamask/safe-event-emitter": "^2.0.0",
     "@metamask/snap-types": "^0.21.0",
     "@metamask/utils": "^3.0.1",
-    "nanoid": "^4.0.0"
+    "nanoid": "^3.1.31"
```
Comment (FrederikBolding) [inline packages/provider/package.json:31]: Why?

File: packages/utils/src/namespace.ts [hunk 3]
```diff
@@ -109,8 +108,7 @@ export const ConnectArgumentsStruct = object({
 });
 export type ConnectArguments = Infer<typeof ConnectArgumentsStruct>;
 
-// TODO: Fix this type.
-export const RequestArgumentsStruct: any = assign(
+export const RequestArgumentsStruct = assign(
```
Comment (FrederikBolding) [inline packages/utils/src/namespace.ts:111]: How did this get fixed?

File: packages/provider/package.json (hunk 2 above)
Comment (Mrtenz) [reply packages/provider/package.json:31]: `4.0.0` is ESM-only. We use `3.1.31` elsewhere.

File: packages/provider/src/MultiChainProvider.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/provider/src/MultiChainProvider.ts:140]: I guess so? It just defaults to `global` on Node.js, and `window` in browsers.

File: packages/utils/src/namespace.ts (hunk 3 above)
Comment (Mrtenz) [reply packages/utils/src/namespace.ts:111]: Not sure, I removed `any` and it works. 🤷

File: packages/provider/src/MultiChainProvider.ts (hunk 1 above)
Comment (FrederikBolding) [reply packages/provider/src/MultiChainProvider.ts:140]: Could we use `globalThis`?

Comment (FrederikBolding) [reply packages/provider/src/MultiChainProvider.ts:140]: May need `declare global`

Comment (Mrtenz) [reply packages/provider/src/MultiChainProvider.ts:140]:
```ts
declare global {
  const ethereum: SnapProvider;
}

globalThis.ethereum;
// TS2339: Property 'ethereum' does not exist on type 'typeof globalThis'.
```

File: packages/provider/jest.config.js [hunk 4]
```diff
@@ -22,7 +22,10 @@ module.exports = {
   // original implementations, between each test. It does not affect mocked
   // modules.
   restoreMocks: true,
-  testEnvironment: 'node',
+  testEnvironment: 'jsdom',
+  testEnvironmentOptions: {
+    customExportConditions: ['node', 'node-addons'],
```
Comment (FrederikBolding) [inline packages/provider/jest.config.js:27]: Can we remove this now?

Comment (Mrtenz) [reply packages/provider/jest.config.js:27]: No, we need that regardless of environment AFAIK.

**Pull request merged**